go/common/entity/entity_test.go

```diff
@@ -0,0 +1,60 @@
+package entity
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/oasisprotocol/oasis-core/go/common/cbor"
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/signature"
+	memorySigner "github.com/oasisprotocol/oasis-core/go/common/crypto/signature/signers/memory"
+)
+
+func TestEntityDescriptorVersioning(t *testing.T) {
+	require := require.New(t)
+
+	type EntityV1 struct {
+		Entity
+		AllowEntitySignedNodes bool `json:"allow_entity_signed_nodes,omitempty"`
+	}
+
+	k1 := memorySigner.NewTestSigner("test entity v1")
+	k1n1 := memorySigner.NewTestSigner("test entity v1 node 1")
+	k1n2 := memorySigner.NewTestSigner("test entity v1 node 2")
+	var ev1 EntityV1
+	ev1.ID = k1.Public()
+	ev1.Nodes = []signature.PublicKey{k1n1.Public(), k1n2.Public()}
+	ev1.V = 1
```

Comment:
Could use `ev1.Versioned = cbor.NewVersioned(1)`. Also you could just do `ev1 := EntityV1{ ...` above.

Reply:
Compiler says no:
```
common/entity/entity_test.go:25:3: cannot use promoted field Entity.Versioned in struct literal of type EntityV1
common/entity/entity_test.go:26:3: cannot use promoted field Entity.ID in struct literal of type EntityV1
common/entity/entity_test.go:27:3: cannot use promoted field Entity.Nodes in struct literal of type EntityV1
```
:D

Reply:
:disappointed: